Web_Development/FrontEnd/CSS_Buttons/readme.md

```diff
@@ -0,0 +1,214 @@
+Syntax:
+.disabled {
+  opacity: 0.6;
+  cursor: not-allowed;
+}
+
+9.Wider Button: By default, the size of the button is determined by its text content (as wide as its content). Use the “width” property to change the width of a button.
+Syntax: .button1 {width: 250px;}
+         OR
+.button2 {width: 50%;}
+
+10.Grouped Buttons: Remove margins and add float:left to each button to create a button group.
+Syntax: .button {
+  float: left;
+}
+
+11.Grouped buttons-Bordered: Use the “border” property and remove margins by adding “float:left” to each button to create button group.
+Syntax:
+.button {
+  float: left;
+  border: 1px solid green;
+}
+
+12.Vertical Button Group: Use display:block instead of float:left to group the buttons below each other, instead of side by side.
+Syntax:
+.button {
+  display: block;
+}
+
+
+
+
+Now, Let’s give some animation to our buttons by Vanilla CSS
+
+1.ADD ARROW ON HOVER
+Code:
+<!DOCTYPE html>
+<html>
+<head>
+<style>
+.button {
+  display: inline-block;
+  border-radius: 4px;
+  background-color: #f4511e;
+  border: none;
+  color: #FFFFFF;
+  text-align: center;
+  font-size: 28px;
+  padding: 20px;
+  width: 200px;
+  transition: all 0.5s;
+  cursor: pointer;
+  margin: 5px;
+}
+
+.button span {
+  cursor: pointer;
+  display: inline-block;
+  position: relative;
+  transition: 0.5s;
+}
+
+.button span:after {
+  content: '\00bb';
+  position: absolute;
+  opacity: 0;
+  top: 0;
+  right: -20px;
+  transition: 0.5s;
+}
+
+.button:hover span {
+  padding-right: 25px;
+}
+
+.button:hover span:after {
+  opacity: 1;
+  right: 0;
+}
+</style>
+</head>
+<body>
+
+<h2>Animated Button</h2>
+
+<button class="button" style="vertical-align:middle"><span>Hover </span></button>
+
+</body>
+</html>
+
+
+
+3.ADD RIPPLE EFFECT ON HOVER:
+
+Code:
+<!DOCTYPE html>
+<html>
+<head>
+<style>
+.button {
+  position: relative;
+  background-color: #4CAF50;
+  border: none;
+  font-size: 28px;
+  color: #FFFFFF;
+  padding: 20px;
+  width: 200px;
+  text-align: center;
+  transition-duration: 0.4s;
+  text-decoration: none;
+  overflow: hidden;
+  cursor: pointer;
+}
+
+.button:after {
+  content: "";
+  background: #f1f1f1;
+  display: block;
+  position: absolute;
+  padding-top: 300%;
+  padding-left: 350%;
+  margin-left: -20px !important;
+  margin-top: -120%;
+  opacity: 0;
+  transition: all 0.8s
+}
+
+.button:active:after {
+  padding: 0;
+  margin: 0;
+  opacity: 1;
+  transition: 0s
+}
+</style>
+</head>
+<body>
+
+<h2>Animated Button - Ripple Effect</h2>
+
+<button class="button">Click Me</button>
+
+</body>
+</html>
+
+
+
+TRICK:
+There are many websites in which you specify what properties or designs you want with your buttons and you can have code for it. Ready to use, just in a flash. One such suggested web are as follows:
+https://css3buttongenerator.com/
+
```

Comment:
Space on top and make it link like this: [CSS Button Generator](https://css3buttongenerator.com/)